api/audit-trail-gw/src/database/channel-info.ts

```diff
@@ -10,9 +10,14 @@ export const getChannelInfo = async (channelAddress: string): Promise<ChannelInf
 
 export const searchChannelInfo = async (channelInfoSearch: ChannelInfoSearch): Promise<ChannelInfoPersistence[]> => {
 	const regex = (text: string) => text && new RegExp(text, 'i');
-	const { authorId, name, created, latestMessage, topicType, topicSource, limit, index, ascending } = channelInfoSearch;
+	const { authorId, subscriberId, requestedSubscriptionId, name, created, latestMessage, topicType, topicSource, limit, index, ascending } = channelInfoSearch;
+
+	const author = MongoDbService.getPlainObject({ authorId: regex(authorId)});
+	const subscriberIds = { subscriberIds: subscriberId ? { $elemMatch: { $eq: subscriberId } } : undefined };
+	const requestedSubscriptionIds = { requestedSubscriptionIds: requestedSubscriptionId ? { $elemMatch: { $eq: requestedSubscriptionId } } : undefined };
+	const idFilterValues = [author, subscriberIds, requestedSubscriptionIds].filter(object => Object.values(object)[0] != undefined);
```

Comment:
Please rename `object` to `filter`
And I don't think `Object.values(filter)[0]` is necessary cant you just use `filter` to check so 

`const idFilterValues = [author, subscriberIds, requestedSubscriptionIds].filter(filter => filter != undefined);`

Reply:
I will adjust this please have a look afterwards

Reply:
@JuriBog adjusted